packages/shared/src/graphql/squads.ts

```diff
@@ -609,8 +608,72 @@ export interface SourcePostModeration extends Partial<PostRequestContent> {
   status: SourcePostModerationStatus;
   reason?: PostModerationReason;
   moderatorMessage?: string;
+  sourceId: Source['id'];
 }
 
+const SOURCE_POST_MODERATION_FRAGMENT = gql`
+  fragment SourcePostModerationFragment on SourcePostModeration {
+    id
+    title
+    sourceId
+    status
+    rejectionReason
+    moderatorMessage
+    type
+    title
+    titleHtml
+    content
+    contentHtml
+    image
+    createdAt
+    createdBy {
+      ...UserAuthor
+    }
+    moderatedBy {
+      ...UserAuthor
+    }
+  }
+  ${USER_AUTHOR_FRAGMENT}
+`;
+
+export const SQUAD_PENDING_POSTS_QUERY = gql`
+  query sourcePostModerations($sourceId: ID!, $status: [String]) {
+    sourcePostModerations(sourceId: $sourceId, status: $status) {
+      edges {
+        node {
+          ...SourcePostModerationFragment
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+  ${SOURCE_POST_MODERATION_FRAGMENT}
+`;
+
+export const SQUAD_MODERATE_POST_MUTATION = gql`
+  mutation ModerateSourcePost(
+    $postIds: [ID]!
+    $status: String
+    $sourceId: ID!
+    $rejectionReason: String
```

Comment:
Reason should be required when rejecting if I'm not mistaken.

```suggestion
    $rejectionReason: String!
```